src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp

```diff
@@ -107,15 +106,21 @@ void TypeLookupReplyListener::process_reply()
                 {
                     case TypeLookup_getTypes_HashId:
                     {
-                        check_get_types_reply(reply.header().relatedRequestId(),
-                                reply.return_value().getType().result(), reply.header().relatedRequestId());
+                        if (RETCODE_OK == reply.return_value().getType()._d())
+                        {
+                            check_get_types_reply(reply.header().relatedRequestId(),
+                                    reply.return_value().getType().result(), reply.header().relatedRequestId());
+                        }
                         break;
                     }
                     case TypeLookup_getDependencies_HashId:
                     {
-                        check_get_type_dependencies_reply(
-                            reply.header().relatedRequestId(), replies_queue_.front().type_server,
-                            reply.return_value().getTypeDependencies().result());
+                        if (RETCODE_OK == reply.return_value().getTypeDependencies()._d())
```

Comment:
What should we do if the discriminator is not `RETCODE_OK`?
Because in that case, the request_id will be kept in the `async_get_type_requests_` collection.

Should we retry the request?
If yes, how many times?
If not, should we notify of the error to any procedure waiting on the type to be complete?